cpg_workflows/stages/gvcf_qc.py

```diff
@@ -26,82 +26,90 @@
 
 
 @stage(required_stages=Genotype)
-class GvcfQC(SampleStage):
+class GvcfQC(SequencingGroupStage):
     """
     Calling tools that process GVCF for QC purposes.
     """
 
-    def expected_outputs(self, sample: Sample) -> dict[str, Path]:
+    def expected_outputs(self, sequencing_group: SequencingGroup) -> dict[str, Path]:
         """
         Generate a GVCF and corresponding TBI index, as well as QC.
         """
         outs: dict[str, Path] = {}
         if get_config()['workflow'].get('skip_qc', False) is False:
-            qc_prefix = sample.dataset.prefix() / 'qc' / sample.id
+            qc_prefix = sequencing_group.dataset.prefix() / 'qc' / sequencing_group.id
             outs |= {
                 'qc_summary': to_path(f'{qc_prefix}.variant_calling_summary_metrics'),
                 'qc_detail': to_path(f'{qc_prefix}.variant_calling_detail_metrics'),
             }
         return outs
 
-    def queue_jobs(self, sample: Sample, inputs: StageInput) -> StageOutput | None:
+    def queue_jobs(
+        self, sequencing_group: SequencingGroup, inputs: StageInput
+    ) -> StageOutput | None:
         """
         Use function from the jobs module
         """
-        gvcf_path = inputs.as_path(sample, Genotype, 'gvcf')
+        gvcf_path = inputs.as_path(sequencing_group, Genotype, 'gvcf')
 
         j = vcf_qc(
             b=get_batch(),
             vcf_or_gvcf=GvcfPath(gvcf_path).resource_group(get_batch()),
             is_gvcf=True,
-            job_attrs=self.get_job_attrs(sample),
-            output_summary_path=self.expected_outputs(sample)['qc_summary'],
-            output_detail_path=self.expected_outputs(sample)['qc_detail'],
-            overwrite=sample.forced,
+            job_attrs=self.get_job_attrs(sequencing_group),
+            output_summary_path=self.expected_outputs(sequencing_group)['qc_summary'],
+            output_detail_path=self.expected_outputs(sequencing_group)['qc_detail'],
+            overwrite=sequencing_group.forced,
+        )
+        return self.make_outputs(
+            sequencing_group, data=self.expected_outputs(sequencing_group), jobs=[j]
         )
-        return self.make_outputs(sample, data=self.expected_outputs(sample), jobs=[j])
 
 
 @stage(required_stages=Genotype)
-class GvcfHappy(SampleStage):
+class GvcfHappy(SequencingGroupStage):
     """
     Run Happy to validate a GVCF for samples where a truth callset is available.
     """
 
-    def expected_outputs(self, sample: Sample) -> Path | None:
+    def expected_outputs(self, sequencing_group: SequencingGroup) -> Path | None:
         """
         Parsed by MultiQC: '*.summary.csv'
         https://multiqc.info/docs/#hap.py
         """
-        if sample.participant_id not in get_config().get('validation', {}).get(
-            'sample_map', {}
-        ):
+        if sequencing_group.participant_id not in get_config().get(
+            'validation', {}
+        ).get('sample_map', {}):
```

Comment:
Are we leaving this configuration key as `sample_map`?

Reply:
I've left this for now, as it appears to be mapping to 'external sample ids'.

Reply:
Fair enough, makes sense.